aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Merger.java

```diff
@@ -60,17 +62,19 @@
      */
     <T extends Model> Completable merge(ModelWithMetadata<T> modelWithMetadata) {
         ModelMetadata metadata = modelWithMetadata.getSyncMetadata();
+        boolean isDelete = Boolean.TRUE.equals(metadata.isDeleted());
         T model = modelWithMetadata.getModel();
 
-        final Completable modelModification;
-        if (Boolean.TRUE.equals(metadata.isDeleted())) {
-            modelModification = delete(model);
-        } else {
-            modelModification = save(model);
-        }
-
-        return modelModification
+        // Check if there is a pending mutation for this model, in the outbox.
+        return mutationOutbox.hasPendingMutation(model.getId())
+            // If there is *not* a pending mutation, we will proceed. Otherwise, not.
+            .filter(value -> !Boolean.TRUE.equals(value))
```

Comment:
Same deal. There are three possibilities, null, true, false. I only want to catch one of these, in a null-safe way.

Reply:
`Objects.nonNull` is for API >= 24, but otherwise could also write this like:

```
    .filter(Objects::nonNull)
    .filter(Boolean.TRUE::equals)
```
Would probably do something like that on the backend with the Java8 stream APIs (similar to Rx `filter`.)